The Windows test errored on `C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift-argument-parser\Sources\ArgumentParser\Validators\UniqueNamesValidator.swift:55:31: error: cannot convert value of type '[Dictionary<String, Int>.Element]' (aka 'Array<(key: String, value: Int)>') to expected argument type '[String : Int]'`

Looking at that line in ArgumentParser, it looks like the wrong overload of `filter()` was picked by the compiler. It's clearly a source compatibility issue.

--- a/stdlib/public/core/Dictionary.swift
+++ b/stdlib/public/core/Dictionary.swift
@@ -926,12 +926,30 @@ extension Dictionary {
   ///   this dictionary.
   ///
   /// - Complexity: O(*n*), where *n* is the length of the dictionary.
-  @inlinable
-  public func mapValues<T>(
-    _ transform: (Value) throws -> T
-  ) rethrows -> Dictionary<Key, T> {
+  @_alwaysEmitIntoClient
+  public func mapValues<T, E: Error>(
+    _ transform: (Value) throws(E) -> T
+  ) throws(E) -> Dictionary<Key, T> {
     return try Dictionary<Key, T>(_native: _variant.mapValues(transform))
   }
+
+#if !$Embedded
+  // ABI-only entrypoint for the rethrows version of mapValues, which has been
+  // superseded by the typed-throws version. Expressed as "throws", which is
+  // ABI-compatible with "rethrows".
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  @abi(
+    func mapValues<T>(
+      _ transform: (Value) throws -> T
+    ) rethrows -> Dictionary<Key, T>
+  )
+  internal func __rethrows_mapValues<T>(
+    _ transform: (Value) throws -> T
+  ) throws -> Dictionary<Key, T> {
+    try mapValues(transform)
+  }
+#endif
 
   /// Returns a new dictionary containing only the key-value pairs that have
   /// non-`nil` values as the result of transformation by the given closure.

--- a/stdlib/public/core/Dictionary.swift
+++ b/stdlib/public/core/Dictionary.swift
@@ -531,13 +531,13 @@ public struct Dictionary<Key: Hashable, Value> {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     init<S: Sequence>(
       _ keysAndValues: __owned S,
       uniquingKeysWith combine: (Value, Value) throws -> Value
-    ) rethrows where S.Element == (Key, Value)
+    ) throws where S.Element == (Key, Value)
   )
+  @usableFromInline
   internal init<S: Sequence>(
     __rethrows_keysAndValues keysAndValues: __owned S,
     uniquingKeysWith combine: (Value, Value) throws -> Value
@@ -581,13 +581,13 @@ public struct Dictionary<Key: Hashable, Value> {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     init<S: Sequence>(
       grouping values: __owned S,
       by keyForValue: (S.Element) throws -> Key
-    ) rethrows where Value == [S.Element]
+    ) throws where Value == [S.Element]
   )
+  @usableFromInline
   internal init<S: Sequence>(
     __rethrows_grouping values: __owned S,
     by keyForValue: (S.Element) throws -> Key
@@ -960,12 +960,12 @@ extension Dictionary {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     func mapValues<T>(
       _ transform: (Value) throws -> T
-    ) rethrows -> Dictionary<Key, T>
+    ) throws -> Dictionary<Key, T>
   )
+  @usableFromInline
   internal func __rethrows_mapValues<T>(
     _ transform: (Value) throws -> T
   ) throws -> Dictionary<Key, T> {
@@ -1094,13 +1094,13 @@ extension Dictionary {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     mutating func merge<S: Sequence>(
       _ other: __owned S,
       uniquingKeysWith combine: (Value, Value) throws -> Value
-    ) rethrows where S.Element == (Key, Value)
+    ) throws where S.Element == (Key, Value)
   )
+  @usableFromInline
   internal mutating func __rethrows_merge<S: Sequence>(
     _ other: __owned S,
     uniquingKeysWith combine: (Value, Value) throws -> Value
@@ -1150,13 +1150,13 @@ extension Dictionary {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     mutating func merge(
       _ other: __owned [Key: Value],
       uniquingKeysWith combine: (Value, Value) throws -> Value
-    ) rethrows
+    ) throws
   )
+  @usableFromInline
   internal mutating func __rethrows_merge_dictionary(
     _ other: __owned [Key: Value],
     uniquingKeysWith combine: (Value, Value) throws -> Value
@@ -1208,13 +1208,13 @@ extension Dictionary {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     __consuming func merging<S: Sequence>(
       _ other: __owned S,
       uniquingKeysWith combine: (Value, Value) throws -> Value
-    ) rethrows -> [Key: Value] where S.Element == (Key, Value)
+    ) throws -> [Key: Value] where S.Element == (Key, Value)
   )
+  @usableFromInline
   internal __consuming func __rethrows_merging<S: Sequence>(
     _ other: __owned S,
     uniquingKeysWith combine: (Value, Value) throws -> Value
@@ -1268,13 +1268,13 @@ extension Dictionary {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     __consuming func merging(
       _ other: __owned [Key: Value],
       uniquingKeysWith combine: (Value, Value) throws -> Value
-    ) rethrows -> [Key: Value]
+    ) throws -> [Key: Value]
   )
+  @usableFromInline
   internal __consuming func __rethrows_merging_dictionary(
     _ other: __owned [Key: Value],
     uniquingKeysWith combine: (Value, Value) throws -> Value

--- a/stdlib/public/core/Dictionary.swift
+++ b/stdlib/public/core/Dictionary.swift
@@ -533,16 +533,36 @@ public struct Dictionary<Key: Hashable, Value> {
   ///   - combine: A closure that is called with the values for any duplicate
   ///     keys that are encountered. The closure returns the desired value for
   ///     the final dictionary.
-  @inlinable
-  public init<S: Sequence>(
+  @_alwaysEmitIntoClient
+  public init<S: Sequence, E: Error>(
     _ keysAndValues: __owned S,
-    uniquingKeysWith combine: (Value, Value) throws -> Value
-  ) rethrows where S.Element == (Key, Value) {
+    uniquingKeysWith combine: (Value, Value) throws(E) -> Value
+  ) throws(E) where S.Element == (Key, Value) {
     var native = _NativeDictionary<Key, Value>(
       capacity: keysAndValues.underestimatedCount)
     try native.merge(keysAndValues, isUnique: true, uniquingKeysWith: combine)
     self.init(_native: native)
   }
+
+#if !$Embedded
+  // ABI-only entrypoint for the rethrows version of init, which has been
+  // superseded by the typed-throws version. Expressed as "throws", which is
+  // ABI-compatible with "rethrows".
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  @abi(
+    init<S: Sequence>(
+      _ keysAndValues: __owned S,
+      uniquingKeysWith combine: (Value, Value) throws -> Value
+    ) rethrows where S.Element == (Key, Value)
+  )
+  internal init<S: Sequence>(
+    __rethrows_keysAndValues keysAndValues: __owned S,
+    uniquingKeysWith combine: (Value, Value) throws -> Value
+  ) throws where S.Element == (Key, Value) {
+    try self.init(keysAndValues, uniquingKeysWith: combine)
+  }
+#endif
 
   /// Creates a new dictionary whose keys are the groupings returned by the
   /// given closure and whose values are arrays of the elements that returned
@@ -566,13 +586,33 @@ public struct Dictionary<Key: Hashable, Value> {
   ///   - values: A sequence of values to group into a dictionary.
   ///   - keyForValue: A closure that returns a key for each element in
   ///     `values`.
-  @inlinable
-  public init<S: Sequence>(
+  @_alwaysEmitIntoClient
+  public init<S: Sequence, E: Error>(
     grouping values: __owned S,
-    by keyForValue: (S.Element) throws -> Key
-  ) rethrows where Value == [S.Element] {
+    by keyForValue: (S.Element) throws(E) -> Key
+  ) throws(E) where Value == [S.Element] {
     try self.init(_native: _NativeDictionary(grouping: values, by: keyForValue))
   }
+
+#if !$Embedded
+  // ABI-only entrypoint for the rethrows version of init, which has been
+  // superseded by the typed-throws version. Expressed as "throws", which is
+  // ABI-compatible with "rethrows".
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  @abi(
+    init<S: Sequence>(
+      grouping values: __owned S,
+      by keyForValue: (S.Element) throws -> Key
+    ) rethrows where Value == [S.Element]
+  )
+  internal init<S: Sequence>(
+    __rethrows_grouping values: __owned S,
+    by keyForValue: (S.Element) throws -> Key
+  ) throws where Value == [S.Element] {
+    try self.init(grouping: values, by: keyForValue)
+  }
+#endif
 }
 
 //

--- a/stdlib/public/core/DictionaryBridging.swift
+++ b/stdlib/public/core/DictionaryBridging.swift
@@ -572,21 +572,23 @@ extension __CocoaDictionary {
     return result
   }
 
+#if !$Embedded
   // ABI-only entrypoint for the rethrows version of mapValues, which has been
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     func mapValues<Key: Hashable, Value, T>(
       _ transform: (Value) throws -> T
-    ) rethrows -> _NativeDictionary<Key, T>
+    ) throws -> _NativeDictionary<Key, T>
   )
+  @usableFromInline
   internal func __rethrows_mapValues<Key: Hashable, Value, T>(
     _ transform: (Value) throws -> T
   ) throws -> _NativeDictionary<Key, T> {
     try mapValues(transform)
   }
+#endif
 }
 
 extension __CocoaDictionary {

--- a/stdlib/public/core/DictionaryBridging.swift
+++ b/stdlib/public/core/DictionaryBridging.swift
@@ -559,10 +559,10 @@ extension __CocoaDictionary: _DictionaryBuffer {
 }
 
 extension __CocoaDictionary {
-  @inlinable
-  internal func mapValues<Key: Hashable, Value, T>(
-    _ transform: (Value) throws -> T
-  ) rethrows -> _NativeDictionary<Key, T> {
+  @_alwaysEmitIntoClient
+  internal func mapValues<Key: Hashable, Value, T, E: Error>(
+    _ transform: (Value) throws(E) -> T
+  ) throws(E) -> _NativeDictionary<Key, T> {
     var result = _NativeDictionary<Key, T>(capacity: self.count)
     for (cocoaKey, cocoaValue) in self {
       let key = _forceBridgeFromObjectiveC(cocoaKey, Key.self)
@@ -570,6 +570,22 @@ extension __CocoaDictionary {
       try result.insertNew(key: key, value: transform(value))
     }
     return result
+  }
+
+  // ABI-only entrypoint for the rethrows version of mapValues, which has been
+  // superseded by the typed-throws version. Expressed as "throws", which is
+  // ABI-compatible with "rethrows".
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  @abi(
+    func mapValues<Key: Hashable, Value, T>(
+      _ transform: (Value) throws -> T
+    ) rethrows -> _NativeDictionary<Key, T>
+  )
+  internal func __rethrows_mapValues<Key: Hashable, Value, T>(
+    _ transform: (Value) throws -> T
+  ) throws -> _NativeDictionary<Key, T> {
+    try mapValues(transform)
   }
 }
 

--- a/stdlib/public/core/DictionaryVariant.swift
+++ b/stdlib/public/core/DictionaryVariant.swift
@@ -452,10 +452,10 @@ extension Dictionary._Variant {
 }
 
 extension Dictionary._Variant {
-  @inlinable
-  internal func mapValues<T>(
-    _ transform: (Value) throws -> T
-  ) rethrows -> _NativeDictionary<Key, T> {
+  @_alwaysEmitIntoClient
+  internal func mapValues<T, E: Error>(
+    _ transform: (Value) throws(E) -> T
+  ) throws(E) -> _NativeDictionary<Key, T> {
 #if _runtime(_ObjC)
     guard isNative else {
       return try asCocoa.mapValues(transform)
@@ -463,6 +463,24 @@ extension Dictionary._Variant {
 #endif
     return try asNative.mapValues(transform)
   }
+
+#if !$Embedded
+  // ABI-only entrypoint for the rethrows version of mapValues, which has been
+  // superseded by the typed-throws version. Expressed as "throws", which is
+  // ABI-compatible with "rethrows".
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  @abi(
+    func mapValues<T>(
+      _ transform: (Value) throws -> T
+    ) rethrows -> _NativeDictionary<Key, T>
+  )
+  internal func __rethrows_mapValues<T>(
+    _ transform: (Value) throws -> T
+  ) throws -> _NativeDictionary<Key, T> {
+    try mapValues(transform)
+  }
+#endif
 
   @inlinable
   internal mutating func merge<S: Sequence>(

--- a/stdlib/public/core/DictionaryVariant.swift
+++ b/stdlib/public/core/DictionaryVariant.swift
@@ -469,12 +469,12 @@ extension Dictionary._Variant {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     func mapValues<T>(
       _ transform: (Value) throws -> T
-    ) rethrows -> _NativeDictionary<Key, T>
+    ) throws -> _NativeDictionary<Key, T>
   )
+  @usableFromInline
   internal func __rethrows_mapValues<T>(
     _ transform: (Value) throws -> T
   ) throws -> _NativeDictionary<Key, T> {
@@ -510,13 +510,13 @@ extension Dictionary._Variant {
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     mutating func merge<S: Sequence>(
       _ keysAndValues: __owned S,
       uniquingKeysWith combine: (Value, Value) throws -> Value
-    ) rethrows where S.Element == (Key, Value)
+    ) throws where S.Element == (Key, Value)
   )
+  @usableFromInline
   internal mutating func __rethrows_merge<S: Sequence>(
     _ keysAndValues: __owned S,
     uniquingKeysWith combine: (Value, Value) throws -> Value

--- a/stdlib/public/core/NativeDictionary.swift
+++ b/stdlib/public/core/NativeDictionary.swift
@@ -872,12 +872,12 @@ extension _NativeDictionary { // High-level operations
   }
   #endif
 
-  @inlinable
+  @_alwaysEmitIntoClient
   @inline(__always)
-  internal init<S: Sequence>(
+  internal init<S: Sequence, E: Error>(
     grouping values: __owned S,
-    by keyForValue: (S.Element) throws -> Key
-  ) rethrows where Value == [S.Element] {
+    by keyForValue: (S.Element) throws(E) -> Key
+  ) throws(E) where Value == [S.Element] {
     self.init()
     for value in values {
       let key = try keyForValue(value)
@@ -889,6 +889,26 @@ extension _NativeDictionary { // High-level operations
       }
     }
   }
+
+#if !$Embedded
+  // ABI-only entrypoint for the rethrows version of init, which has been
+  // superseded by the typed-throws version. Expressed as "throws", which is
+  // ABI-compatible with "rethrows".
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  @abi(
+    init<S: Sequence>(
+      grouping values: __owned S,
+      by keyForValue: (S.Element) throws -> Key
+    ) rethrows where Value == [S.Element]
+  )
+  internal init<S: Sequence>(
+    __rethrows_grouping values: __owned S,
+    by keyForValue: (S.Element) throws -> Key
+  ) throws where Value == [S.Element] {
+    try self.init(grouping: values, by: keyForValue)
+  }
+#endif
 
   @_alwaysEmitIntoClient
   internal func filter(

--- a/stdlib/public/core/NativeDictionary.swift
+++ b/stdlib/public/core/NativeDictionary.swift
@@ -755,10 +755,10 @@ extension _NativeDictionary { // Deletion
 }
 
 extension _NativeDictionary { // High-level operations
-  @inlinable
-  internal func mapValues<T>(
-    _ transform: (Value) throws -> T
-  ) rethrows -> _NativeDictionary<Key, T> {
+  @_alwaysEmitIntoClient
+  internal func mapValues<T, E: Error>(
+    _ transform: (Value) throws(E) -> T
+  ) throws(E) -> _NativeDictionary<Key, T> {
     let resultStorage = unsafe _DictionaryStorage<Key, T>.copy(original: _storage)
     unsafe _internalInvariant(resultStorage._seed == _storage._seed)
     let result = unsafe _NativeDictionary<Key, T>(resultStorage)
@@ -772,6 +772,24 @@ extension _NativeDictionary { // High-level operations
     }
     return result
   }
+
+#if !$Embedded
+  // ABI-only entrypoint for the rethrows version of mapValues, which has been
+  // superseded by the typed-throws version. Expressed as "throws", which is
+  // ABI-compatible with "rethrows".
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  @abi(
+    func mapValues<T>(
+      _ transform: (Value) throws -> T
+    ) rethrows -> _NativeDictionary<Key, T>
+  )
+  internal func __rethrows_mapValues<T>(
+    _ transform: (Value) throws -> T
+  ) throws -> _NativeDictionary<Key, T> {
+    try mapValues(transform)
+  }
+#endif
 
   @inlinable
   internal mutating func merge<S: Sequence>(

--- a/stdlib/public/core/NativeDictionary.swift
+++ b/stdlib/public/core/NativeDictionary.swift
@@ -778,12 +778,12 @@ extension _NativeDictionary { // High-level operations
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     func mapValues<T>(
       _ transform: (Value) throws -> T
-    ) rethrows -> _NativeDictionary<Key, T>
+    ) throws -> _NativeDictionary<Key, T>
   )
+  @usableFromInline
   internal func __rethrows_mapValues<T>(
     _ transform: (Value) throws -> T
   ) throws -> _NativeDictionary<Key, T> {
@@ -835,14 +835,14 @@ extension _NativeDictionary { // High-level operations
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     mutating func merge<S: Sequence>(
       _ keysAndValues: __owned S,
       isUnique: Bool,
       uniquingKeysWith combine: (Value, Value) throws -> Value
-    ) rethrows where S.Element == (Key, Value)
+    ) throws where S.Element == (Key, Value)
   )
+  @usableFromInline
   internal mutating func __rethrows_merge<S: Sequence>(
     _ keysAndValues: __owned S,
     isUnique: Bool,
@@ -875,13 +875,13 @@ extension _NativeDictionary { // High-level operations
   // superseded by the typed-throws version. Expressed as "throws", which is
   // ABI-compatible with "rethrows".
   @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
-  @usableFromInline
   @abi(
     init<S: Sequence>(
       grouping values: __owned S,
       by keyForValue: (S.Element) throws -> Key
-    ) rethrows where Value == [S.Element]
+    ) throws where Value == [S.Element]
   )
+  @usableFromInline
   internal init<S: Sequence>(
     __rethrows_grouping values: __owned S,
     by keyForValue: (S.Element) throws -> Key

--- a/test/api-digester/Outputs/stability-stdlib-source-base.swift.expected
+++ b/test/api-digester/Outputs/stability-stdlib-source-base.swift.expected
@@ -361,6 +361,8 @@ Func ContiguousArray.withUnsafeBufferPointer(_:) is now without rethrows
 Func ContiguousArray.withUnsafeMutableBufferPointer(_:) has generic signature change from <Element, R> to <Element, R, E where E : Swift.Error>
 Func ContiguousArray.withUnsafeMutableBufferPointer(_:) has parameter 0 type change from (inout Swift.UnsafeMutableBufferPointer<Element>) throws -> R to (inout Swift.UnsafeMutableBufferPointer<Element>) throws(E) -> R
 Func ContiguousArray.withUnsafeMutableBufferPointer(_:) is now without rethrows
+Func Dictionary.mapValues(_:) has generic signature change from <Key, Value, T where Key : Swift.Hashable> to <Key, Value, T, E where Key : Swift.Hashable, E : Swift.Error>
+Func Dictionary.mapValues(_:) is now without rethrows
 
 // Adoption of @_addressableForDependencies
 Struct CollectionOfOne is now with @_addressableForDependencies

--- a/test/api-digester/Outputs/stability-stdlib-source-base.swift.expected
+++ b/test/api-digester/Outputs/stability-stdlib-source-base.swift.expected
@@ -369,6 +369,10 @@ Func Dictionary.merge(_:uniquingKeysWith:) is now without rethrows
 Func Dictionary.merging(_:uniquingKeysWith:) has generic signature change from <Key, Value where Key : Swift.Hashable> to <Key, Value, E where Key : Swift.Hashable, E : Swift.Error>
 Func Dictionary.merging(_:uniquingKeysWith:) has generic signature change from <Key, Value, S where Key : Swift.Hashable, S : Swift.Sequence, S.Element == (Key, Value)> to <Key, Value, S, E where Key : Swift.Hashable, S : Swift.Sequence, E : Swift.Error, S.Element == (Key, Value)>
 Func Dictionary.merging(_:uniquingKeysWith:) is now without rethrows
+Constructor Dictionary.init(_:uniquingKeysWith:) has generic signature change from <Key, Value, S where Key : Swift.Hashable, S : Swift.Sequence, S.Element == (Key, Value)> to <Key, Value, S, E where Key : Swift.Hashable, S : Swift.Sequence, E : Swift.Error, S.Element == (Key, Value)>
+Constructor Dictionary.init(_:uniquingKeysWith:) is now without rethrows
+Constructor Dictionary.init(grouping:by:) has generic signature change from <Key, Value, S where Key : Swift.Hashable, Value == [S.Element], S : Swift.Sequence> to <Key, Value, S, E where Key : Swift.Hashable, Value == [S.Element], S : Swift.Sequence, E : Swift.Error>
+Constructor Dictionary.init(grouping:by:) is now without rethrows
 
 // Adoption of @_addressableForDependencies
 Struct CollectionOfOne is now with @_addressableForDependencies

--- a/test/api-digester/stability-stdlib-abi-without-asserts.test
+++ b/test/api-digester/stability-stdlib-abi-without-asserts.test
@@ -819,6 +819,9 @@ Func _SliceBuffer.withUnsafeMutableBufferPointer(_:) has mangled name changing f
 
 // Functions changed for typed throws
 Constructor Array.init(_unsafeUninitializedCapacity:initializingWith:) has been removed
+Constructor Dictionary.init(_:uniquingKeysWith:) has been removed
+Constructor Dictionary.init(grouping:by:) has been removed
+Constructor _NativeDictionary.init(grouping:by:) has been removed
 Func Array.withUnsafeBufferPointer(_:) has been removed
 Func ArraySlice.withUnsafeBufferPointer(_:) has been removed
 Func ContiguousArray.withUnsafeBufferPointer(_:) has been removed

--- a/test/api-digester/stability-stdlib-abi-without-asserts.test
+++ b/test/api-digester/stability-stdlib-abi-without-asserts.test
@@ -822,6 +822,11 @@ Constructor Array.init(_unsafeUninitializedCapacity:initializingWith:) has been 
 Func Array.withUnsafeBufferPointer(_:) has been removed
 Func ArraySlice.withUnsafeBufferPointer(_:) has been removed
 Func ContiguousArray.withUnsafeBufferPointer(_:) has been removed
+Func Dictionary._Variant.mapValues(_:) has been removed
+Func Dictionary.mapValues(_:) has been removed
+Func _NativeDictionary.mapValues(_:) has been removed
+Func __CocoaDictionary.mapValues(_:) has been removed
+
 
 Struct String.Index has added a conformance to an existing protocol CustomDebugStringConvertible
 

